dali/operators/reader/loader/video_loader.cc

```diff
@@ -586,11 +591,21 @@ void VideoLoader::read_file() {
     file.last_frame_ = frame;
     key = (pkt->flags & AV_PKT_FLAG_KEY) != 0;
 
-    // if decoding hasn't produced any frames after providing kStartupFrameTreshold frames,
+    // if decoding hasn't produced any frames after providing kStartupFrameThreshold frames,
     // or we are at next key frame
     if (last_key_frame != -1 &&
-        ((key && last_key_frame != frame) || frame > last_key_frame + kStartupFrameTreshold) &&
+        ((key && last_key_frame != frame && last_key_frame != 0) ||
+          frame > last_key_frame + kStartupFrameThreshold) &&
         dec_status == VidReqStatus::REQ_NOT_STARTED) {
+        if (last_key_frame <= 0) {
+          if (vid_decoder_) {
+            vid_decoder_->finish();
+          }
+          DALI_FAIL("Decoding not started when starting from frame 0, "
+                    "no point in seeking to preceding keyframe which is "
+                    "already 0");
```

Comment:
Is there particular kind of video file or parameter set that triggers this problem? Maybe it's worth mentioning in the message? The current one doesn't seem very helpful for the user as it only refers to the internals of the decoder

Reply:
The faulty sample was provided by the user in https://github.com/NVIDIA/DALI/issues/5155#issuecomment-1837082791, but I don't know how to create such one on our own and I don't think we can add this one to DALI_extra.